### PR TITLE
feat: integrate optax optimizers for JAX backend

### LINKED ISSUE #230: feat: Integrate optax optimizers (Adam, AdaGrad) for JAX backend
## Problem

L-BFGS-B is the only gradient-based optimizer currently used with JAX analytical gradients. L-BFGS-B is a quasi-Newton method designed for smooth, well-conditioned landscapes — it converges fast but has no mechanism to escape local minima. On rugged MM3 landscapes, it gets stuck (579 RMSD on CH3F).

Modern deep learning has developed optimizers specifically designed for non-convex landscapes that are natively compatible with JAX.

## Proposal

Integrate `optax` (JAX's optimizer library) to provide alternative gradient-based optimizers:

- **Adam**: Adaptive learning rates + momentum. The momentum term accumulates gradient history, helping traverse flat regions and escape shallow minima.
- **AdaGrad**: Per-parameter adaptive learning rates. Handles parameters with very different scales (e.g., force constants in mdyn/Å vs angles in degrees).
- **SGD with momentum**: Simple but effective; momentum provides exploration similar to FD noise.
- **Learning rate schedules**: Warm restarts (cosine annealing) periodically reset the learning rate, providing another mechanism to escape local minima.

## Expected Impact

- Adam/AdaGrad may find better minima than L-BFGS-B due to momentum-based exploration
- Per-parameter learning rates handle the scale mismatch between different FF parameter types
- Native JAX implementation — no SciPy interop overhead
- Opens the door to training-style workflows (epochs, batches, schedules)

## Implementation Notes

- `optax` is already a JAX dependency (or trivially installable)
- Need a training loop wrapper since `optax` optimizers are iterative (not `minimize`-style)
- Convergence criterion: monitor objective value plateau or gradient norm threshold
- Key hyperparameters: learning rate, momentum (β1, β2 for Adam), number of steps
- Should integrate with existing q2mm objective function and gradient pipeline

## References

- optax documentation: https://optax.readthedocs.io/
- JAX-MD uses similar patterns: https://github.com/jax-md/jax-md
- DMFF (differentiable FF library) uses JAX autodiff for FF optimization: https://docs.deepmodeling.com/projects/dmff/en/stable/

### PR DESCRIPTION
**Pull request by ericchansen**
## Summary

Integrate [optax](https://optax.readthedocs.io/) adaptive optimizers (Adam, AdaGrad, SGD, AdamW) as an alternative to scipy's L-BFGS-B for JAX-based force field optimization.

**Headline result:** Adam on CH3F MM3 achieves **56.3 cm-1 RMSD** -- 10x better than L-BFGS-B's 579.0, and the best JAX MM3 result across all optimizers and backends.

### Code changes
- `q2mm/optimizers/optax.py` -- `OptaxOptimizer` class with Adam/AdaGrad/SGD/AdamW, LR schedules, bounds clamping
- `q2mm/optimizers/__init__.py` -- lazy import
- `q2mm/optimizers/cycling.py` -- `optax:` prefix dispatch in cycling loop
- `q2mm/diagnostics/benchmark.py` -- optax dispatch in `run_combo()`, colon handling in `benchmark_stem()`
- `q2mm/diagnostics/cli.py` -- 4 optax configs, `--learning-rate` and `--optax-max-steps` CLI args
- `pyproject.toml` -- added `"optax"` to `[jax]` dependency group

### Tests
- 21 unit tests (`test/test_optax.py`)
- 6 integration tests (`test/integration/test_optax.py`)

### Benchmark results (8 new CH3F combos)
| Optimizer | RMSD (harmonic) | RMSD (MM3) |
|-----------|----------------:|-----------:|
| optax:adam | 1000.4 | **56.3** |
| optax:adam+cosine | 999.4 | 60.6 |
| optax:adagrad | 1000.9 | 138.0 |
| optax:sgd | 990.0 | 192.0 (diverged) |
| L-BFGS-B (analytical) | **528.7** | 579.0 |

### Documentation (10 pages updated)
- Tutorial Step 6c, optimization guide Strategy 2, small-molecules table (44 combos), JAX backend page, comparison page, home page, getting-started, architecture, benchmarks index, CHANGELOG

Closes #230

### CODE REVIEW

File: q2mm/optimizers/cycling.py [hunk 1]
```diff
@@ -521,14 +524,25 @@ def run(self) -> LoopResult:
...
-                    verbose=False,
-                )
-                full_result = full_opt.optimize(self.objective)
+                if use_optax:
+                    from q2mm.optimizers.optax import OptaxOptimizer
+
+                    optax_name = self.full_method.split(":", 1)[1]
+                    full_opt = OptaxOptimizer(
+                        optimizer=optax_name,
+                        max_steps=self.full_maxiter,
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/optimizers/cycling.py:538]: Fixed in b8bba05. Cycling now splits the +schedule suffix before passing to OptaxOptimizer, so optax:adam+cosine correctly becomes optimizer="adam", schedule="cosine".

File: docs/tutorial.md [hunk 2]
```diff
@@ -699,6 +699,76 @@ parameters, combining the strengths of both approaches.
...
+    ```
+
+    Expected output:
+
+    ```
+    Method: optax:adam
+    Converged: False (max_steps reached)
+    Score: 192.000 → 56.300
+    Steps: 2000
+    ```
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply docs/tutorial.md:737]: Fixed in b8bba05. Updated expected output to match the actual OptimizationResult.summary() format (Success/Iterations/Evaluations, not Converged/Steps).
